gax-grpc/src/main/java/com/google/api/gax/grpc/GrpcFieldToHeaderCallable.java

```diff
@@ -29,44 +29,33 @@
  */
 package com.google.api.gax.grpc;
 
-import com.google.common.collect.ImmutableMap;
-import io.grpc.CallOptions;
-import io.grpc.Metadata;
-import io.grpc.Metadata.Key;
-import java.util.Collections;
-import java.util.Map;
-
-/** A utility class that provides helper functions to work with custom call options. */
-class CallOptionsUtil {
-  // this is a call option name, not a header name, it is not transferred over the wire
-  private static final CallOptions.Key<Map<Key<String>, String>> DYNAMIC_HEADERS_CALL_OPTION_KEY =
-      CallOptions.Key.of("gax_dynamic_headers", Collections.<Key<String>, String>emptyMap());
-  // this is the header name, it is transferred over the wire
-  static Metadata.Key<String> REQUEST_PARAMS_HEADER_KEY =
-      Metadata.Key.of("x-goog-request-params", Metadata.ASCII_STRING_MARSHALLER);
-
-  private CallOptionsUtil() {}
-
-  static CallOptions putRequestParamsDynamicHeaderOption(
-      CallOptions callOptions, String requestParams) {
-    if (callOptions == null || requestParams.isEmpty()) {
-      return callOptions;
-    }
-
-    Map<Key<String>, String> dynamicHeadersOption =
-        callOptions.getOption(DYNAMIC_HEADERS_CALL_OPTION_KEY);
-
-    // This will fail, if REQUEST_PARAMS_HEADER_KEY is already there
-    dynamicHeadersOption =
-        ImmutableMap.<Key<String>, String>builder()
-            .putAll(dynamicHeadersOption)
-            .put(REQUEST_PARAMS_HEADER_KEY, requestParams)
-            .build();
-
-    return callOptions.withOption(DYNAMIC_HEADERS_CALL_OPTION_KEY, dynamicHeadersOption);
+import com.google.api.core.ApiFuture;
+import com.google.api.gax.rpc.ApiCallContext;
+import com.google.api.gax.rpc.RequestParamsExtractor;
+import com.google.api.gax.rpc.RequestUrlParamsEncoder;
+import com.google.api.gax.rpc.UnaryCallable;
+import com.google.common.base.Preconditions;
+
+public class GrpcFieldToHeaderCallable<RequestT, ResponseT>
```

Comment:
Please consider naming adding `Unary` to the name, so it matches its server streaming counterpart. Also, please consider naming this (and streaming version) to something around "RequestParams", since this is the name for this feature in all other places (the design doc, the actual header name, toolkit).

Reply:
Ok will do.